### PR TITLE
Allow array as value in externals object

### PR DESCRIPTION
**Pull request by RubenVerborgh**
Fixes https://github.com/webpack/webpack/issues/8041.

https://github.com/webpack/webpack/issues/8041#issuecomment-421972351

**What kind of change does this PR introduce?**

bugfix (schema)

**Did you add tests for your changes?**

Yes.

**Does this PR introduce a breaking change?**

No.

**What needs to be documented once your changes are merged?**

The documentation is alright; it already mentions this option:
https://webpack.js.org/configuration/externals/#array
(but that doesn't work yet without this PR)

### CODE REVIEW

File: schemas/WebpackOptions.json [hunk 1]
```diff
@@ -123,6 +123,9 @@
               {
                 "type": "object"
               },
+              {
+                "type": "array"
```
Comment (sokra) [inline schemas/WebpackOptions.json]: Be more specific with which `items` are accepted.

**Comment by RubenVerborgh** [comment]
@sokra Done.

File: test/configCases/externals/optional-externals-array/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+	externals: {
+		external: ["./math", "subtract"]
```
Comment (sokra) [inline test/configCases/externals/optional-externals-array/webpack.config.js]: You can use `["webpack", "version"]` and check if the result is a string.

Checking the error doesn't really check if `subtract` is used or ignored.

Comment (RubenVerborgh) [reply test/configCases/externals/optional-externals-array/webpack.config.js]: I just copied what is in the other `test/configCases/externals/optional-externals-*/webpack.config.js` tests. These tests are just testing the configuration, I would expect there to be other tests that actually verify the functionality.

Comment (RubenVerborgh) [reply test/configCases/externals/optional-externals-array/webpack.config.js]: Just tried `["webpack", "version"]` and it still errors (test passes without alteration). So if that is supposed to work, there's another issue (which we should then follow up on in #8041 I guess).

Comment (sokra) [reply test/configCases/externals/optional-externals-array/webpack.config.js]: Did you follow the CONTRIBUTION instructions about `yarn link && yarn link webpack`?

Comment (RubenVerborgh) [reply test/configCases/externals/optional-externals-array/webpack.config.js]: Followed the instructions, same result. Looks like CI failed due to a random timeout (only AppVeyor failed). Will push commit again.

**Comment by sokra** [comment]
Thanks

**Comment by sokra** [comment]
btw. your test didn't work, because the Array version wasn't really working internally. It works now.

**Comment by RubenVerborgh** [comment]
@sokra Yeah that's what I figured. Thanks for fixing. Good that you insisted on a functionality test.

**Pull request merged**